### PR TITLE
[td-10748]interp

### PR DESCRIPTION
**Pull request by dapan1121**

### CODE REVIEW

File: src/query/src/qExecutor.c [hunk 1]
```diff
@@ -6047,127 +5910,477 @@ static SSDataBlock* doIntervalAgg(void* param, bool* newgroup) {
...
+      }
+    }
+
     STableQueryInfo* pTableQueryInfo = pRuntimeEnv->current;
 
-    setTagValue(pOperator, pTableQueryInfo->pTable, pIntervalInfo->pCtx, pOperator->numOfOutput);
-    setInputDataBlock(pOperator, pIntervalInfo->pCtx, pBlock, pQueryAttr->order.order);
-    setIntervalQueryRange(pRuntimeEnv, &pBlock->info.window, pBlock->info.tid);
+    if (pEveryInfo->groupDone && pTableQueryInfo->groupIndex == pEveryInfo->lastGroupIdx) {
+      assert(pOperator->upstream[0]->notify == NULL);
```
Comment (shenglian-zhou) [inline src/query/src/qExecutor.c:6366]: did not find the location where notify is set to NULL

File: src/inc/taosmsg.h [hunk 2]
```diff
@@ -479,6 +479,7 @@ typedef struct {
   bool        stateWindow;       // state window flag 
 
   STimeWindow window;
+  STimeWindow range;
```
Comment (hjxilinx) [inline src/inc/taosmsg.h]: Could you please add a little bit more comment on this attribute to explain why this attribute is needed?

**Review by hjxilinx — CHANGES_REQUESTED** [review]

File: src/inc/taosmsg.h (hunk 2 above)
Comment (dapan1121) [reply src/inc/taosmsg.h]: ok

**Review by hjxilinx — APPROVED** [review]
I'm OK with this PR.

File: src/query/src/qFill.c [hunk 3]
```diff
@@ -493,12 +494,20 @@ int64_t getNumOfResultsAfterFillGap(SFillInfo* pFillInfo, TSKEY ekey, int32_t ma
   return (numOfRes > maxNumOfRows) ? maxNumOfRows : numOfRes;
 }
 
-int32_t taosGetLinearInterpolationVal(SPoint* point, int32_t outputType, SPoint* point1, SPoint* point2, int32_t inputType) {
-  double v1 = -1, v2 = -1;
+int32_t taosGetLinearInterpolationVal(SPoint* point, int32_t outputType, SPoint* point1, SPoint* point2, int32_t inputType, bool *exceedMax, bool *exceedMin) {
+  double v1 = -1, v2 = -1, vmax = -1, vmin = -1;
   GET_TYPED_DATA(v1, double, inputType, point1->val);
```
Comment (shenglian-zhou) [inline src/query/src/qFill.c:499]: for integer values, keep double calculation at the last step and as much as integer calculation as possible?

Comment (dapan1121) [reply src/query/src/qFill.c:499]: 好像意义不是很大，既然运算过程中肯定要 double保持精度，那就必然有需要转到double的部分。

**Pull request merged**